python/pyspark/broadcast.py

```diff
@@ -80,7 +81,7 @@ def __init__(self, sc=None, value=None, pickle_registry=None, path=None):
 
     def dump(self, value, f):
         try:
-            pickle.dump(value, f, 2)
+            pickle.dump(value, f, protocol)
```

Comment:
can we match our cloudpickle copy to upper version for this accordingly?

Reply:
@HyukjinKwon it seems like the test failure from the report 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/87852/testReport is reproducible even in master branch without my commits, isn't it?